### ISSUE TITLE
About testing result

### ISSUE THREAD

**Issue by wanboyang** [zhangxuying1004/RSTNet#25]
Could you provide the inference captions of COCO test set by using model reported in this paper?
Thanks.

**Comment by wanboyang** [comment]
I use the pretrained model provide in this project to test COCO test set, and eval performance is 'BLEU': [0.806718016311047, 0.6540980786273354, 0.5092295283787919, 0.3898375465182736], 'METEOR': 0.29228558925065895, 'ROUGE': 0.585628568332276, 'CIDEr': 1.3206302796307086.
It seems different from the performance claimed in paper.
Could you provide the inference captions of COCO test set by using the method proposed in this paper.

**Comment by hasky123** [comment]
你好，能否提供一份特征？感激不尽！！

**Comment by wanboyang** [comment]
我使用的特征来自https://github.com/luo3300612/image-captioning-DLCT

在 2022年3月15日星期二，hasky123 ***@***.***> 写道：

**Comment by zhangxuying1004** [comment]
> I use the pretrained model provide in this project to test COCO test set, and eval performance is 'BLEU': [0.806718016311047, 0.6540980786273354, 0.5092295283787919, 0.3898375465182736], 'METEOR': 0.29228558925065895, 'ROUGE': 0.585628568332276, 'CIDEr': 1.3206302796307086. It seems different from the performance claimed in paper. Could you provide the inference captions of COCO test set by using the method proposed in this paper.

关于language model，您有没有使用我上传的的模型参数文件？还是您重新训练的？

**Comment by wanboyang** [comment]
> > I use the pretrained model provide in this project to test COCO test set, and eval performance is 'BLEU': [0.806718016311047, 0.6540980786273354, 0.5092295283787919, 0.3898375465182736], 'METEOR': 0.29228558925065895, 'ROUGE': 0.585628568332276, 'CIDEr': 1.3206302796307086. It seems different from the performance claimed in paper. Could you provide the inference captions of COCO test set by using the method proposed in this paper.
> 
> 关于language model，您有没有使用我上传的的模型参数文件？还是您重新训练的？

我使用的是上传的的模型参数文件

**Comment by wanboyang** [comment]
decoder = TransformerDecoderLayer(len(text_field.vocab), 54, 3, text_field.vocab.stoi['<pad>'], 'Pre-trained_Models/language_context.pth')

**Comment by zhangxuying1004** [comment]
> > > I use the pretrained model provide in this project to test COCO test set, and eval performance is 'BLEU': [0.806718016311047, 0.6540980786273354, 0.5092295283787919, 0.3898375465182736], 'METEOR': 0.29228558925065895, 'ROUGE': 0.585628568332276, 'CIDEr': 1.3206302796307086. It seems different from the performance claimed in paper. Could you provide the inference captions of COCO test set by using the method proposed in this paper.
> > 
> > 
> > 关于language model，您有没有使用我上传的的模型参数文件？还是您重新训练的？
> 
> 我使用的是上传的的模型参数文件

我check一下。

**Comment by wanboyang** [comment]
> > > > I use the pretrained model provide in this project to test COCO test set, and eval performance is 'BLEU': [0.806718016311047, 0.6540980786273354, 0.5092295283787919, 0.3898375465182736], 'METEOR': 0.29228558925065895, 'ROUGE': 0.585628568332276, 'CIDEr': 1.3206302796307086. It seems different from the performance claimed in paper. Could you provide the inference captions of COCO test set by using the method proposed in this paper.
> > > 
> > > 
> > > 关于language model，您有没有使用我上传的的模型参数文件？还是您重新训练的？
> > 
> > 
> > 我使用的是上传的的模型参数文件
> 
> 我check一下。

请问有结果了吗

**Comment by egoistsh** [comment]
你好，能更新代码吗，这个精度确实不对。

**Comment by gjc0824** [comment]
> I use the pretrained model provide in this project to test COCO test set, and eval performance is 'BLEU': [0.806718016311047, 0.6540980786273354, 0.5092295283787919, 0.3898375465182736], 'METEOR': 0.29228558925065895, 'ROUGE': 0.585628568332276, 'CIDEr': 1.3206302796307086. It seems different from the performance claimed in paper. Could you provide the inference captions of COCO test set by using the method proposed in this paper.

Hello Wan！I also use this open source code to train the RSTNet. However, at the end of cross-entropy training stage,  I only get the CIDEr score about 103, is that normal? How much CIDEr score did you get at the end of cross-entropy training stage? How many epochs did you train during RL stage? Did you use other tricks that can tell me?
Thank you very much! I expect your reply.

**Comment by wanboyang** [comment]
> Hello Wan！I also use this open source code to train the RSTNet. However, at the end of cross-entropy training stage, I only get the CIDEr score about 103, is that normal? How much CIDEr score did you get at the end of cross-entropy training stage? How many epochs did you train during RL stage? Did you use other tricks that can tell me?
> Thank you very much! I expect your reply.

I do not train the RSTNet.

**Comment by zhuhaifengaaa** [comment]
> 你好，能更新代码吗，这个精度确实不对。
是啊，这个模型跑的结果比不上论文里面的，我不知道哪出了问题，作者能帮解答一下吗？谢谢了

**Comment by zhangxuying1004** [comment]
代码已更新，下面是测试结果截图。
![image](https://user-images.githubusercontent.com/47649909/167592055-f1d581d8-c1b9-4265-861e-dacb211ef78d.png)

**Comment by zhangxuying1004** [comment]
> > > > > I use the pretrained model provide in this project to test COCO test set, and eval performance is 'BLEU': [0.806718016311047, 0.6540980786273354, 0.5092295283787919, 0.3898375465182736], 'METEOR': 0.29228558925065895, 'ROUGE': 0.585628568332276, 'CIDEr': 1.3206302796307086. It seems different from the performance claimed in paper. Could you provide the inference captions of COCO test set by using the method proposed in this paper.
> > > > 
> > > > 
> > > > 关于language model，您有没有使用我上传的的模型参数文件？还是您重新训练的？
> > > 
> > > 
> > > 我使用的是上传的的模型参数文件
> > 
> > 
> > 我check一下。
> 
> 请问有结果了吗

已更新
